pkg/operator/controller/names.go

```diff
@@ -20,19 +20,63 @@ import (
 	"fmt"
 	"hash"
 	"hash/fnv"
+	"strings"
 
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/discovery"
 
 	operatorv1alpha1 "github.com/openshift/external-dns-operator/api/v1alpha1"
 	operatorconfig "github.com/openshift/external-dns-operator/pkg/operator/config"
 )
 
 const (
 	// ExternalDNSBaseName is the base name for any ExternalDNS resource.
-	ExternalDNSBaseName = "external-dns"
+	ExternalDNSBaseName         = "external-dns"
+	CredentialsRequestNamespace = "openshift-cloud-credential-operator"
+	kind                        = "OpenShiftAPIServer"
+	group                       = "operator.openshift.io"
+	version                     = "v1"
 )
 
+// Returns true if platform is OCP
+func IsOCP(kubeClient discovery.DiscoveryInterface) bool {
+	// Since, CRD for OpenShift API Server was introduced in OCP v4.x we can verify if the current cluster is on OCP v4.x by
+	// ensuring that resource exists against Group(operator.openshift.io), Version(v1) and Kind(OpenShiftAPIServer)
+	// In case it doesn't exist we assume that external dns is running on non OCP 4.x environment
+	resources, err := kubeClient.ServerResourcesForGroupVersion(group + "/" + version)
+	if err != nil {
+		return false
+	}
+
+	for _, apiResource := range resources.APIResources {
+		if apiResource.Kind == kind {
+			return true
+		}
+	}
+	return false
+}
+
+func ExternalDNSCredentialsRequestName(externalDNS *operatorv1alpha1.ExternalDNS) types.NamespacedName {
+	switch externalDNS.Spec.Provider.Type {
+	case operatorv1alpha1.ProviderTypeAWS:
+		return types.NamespacedName{
+			Namespace: CredentialsRequestNamespace,
+			Name:      "externaldns-credentials-request-" + strings.ToLower(string(operatorv1alpha1.ProviderTypeAWS))}
+	case operatorv1alpha1.ProviderTypeAzure:
+		return types.NamespacedName{
+			Namespace: CredentialsRequestNamespace,
+			Name:      "externaldns-credentials-request-" + strings.ToLower(string(operatorv1alpha1.ProviderTypeAzure)),
+		}
+	case operatorv1alpha1.ProviderTypeGCP:
+		return types.NamespacedName{
+			Namespace: CredentialsRequestNamespace,
+			Name:      "externaldns-credentials-request-" + strings.ToLower(string(operatorv1alpha1.ProviderTypeGCP)),
+		}
+	}
+	return types.NamespacedName{}
```

Comment:
Since we are only calling this function when the provider is of type AWS, Azure or GCP:
```suggestion
	return types.NamespacedName{
		Namespace: CredentialsRequestNamespace,
		Name:      "externaldns-credentials-request-" + strings.ToLower(string(externalDNS.Spec.Provider.Type))
	}
```